src/compiler/Stage/InferTypes/GenerateEquations.elm

```diff
@@ -39,8 +40,12 @@ import Stage.InferTypes.TypeEquation exposing (TypeEquation, equals)
 import Transform
 
 
-generateEquations : IdSource -> Typed.Expr -> ( List TypeEquation, IdSource )
-generateEquations idSource ( expr, type_ ) =
+generateEquations : IdSource -> Typed.LocatedExpr -> ( List TypeEquation, IdSource )
+generateEquations idSource located_ =
+    let
+        ( expr, type_ ) =
+            Located.unwrap located_
```

Comment:
no need for `located_` I believe as there's nothing that would shadow the name `located`?